### PR TITLE
feat:  create initial http server setup

### PR DESCRIPTION
**Pull request by emirsh**
create initial http server setup

Signed-off-by: Emir Heidinger <emir.heidinger@securekey.com>

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -6,12 +6,73 @@ SPDX-License-Identifier: Apache-2.0
...
+// GetHTTPHandler returns the main http handler (router) that is used by the server
+func (s *Server) GetHTTPHandler() http.Handler {
+	return s.router
+}
+
+// CollectionsHandler handles the actual service request
+func collectionsHandler(w http.ResponseWriter, r *http.Request) {
+	// getModelReq
+
+	http.Error(w, "", http.StatusNotImplemented)
```
Comment (llorllale) [inline internal/server/server.go]: @emirsh should leave a TODO here as well

Comment (emirsh) [reply internal/server/server.go]: Fixed

File: cmd/hub-store/main.go [hunk 2]
```diff
@@ -6,6 +6,61 @@ SPDX-License-Identifier: Apache-2.0
...
+
 func main() {
-	panic("not implemented yet")
+	InitViperConfig()
+
+	s := server.NewServer(getServerConfig())
+
+	http.Handle("/hub-store", s.GetHTTPHandler())
+
+	if err := http.ListenAndServeTLS(s.Config.Host+":"+strconv.Itoa(s.Config.Port), s.Config.TLSCertificateFile,
```
Comment (rolsonquadras) [inline cmd/hub-store/main.go]: Is there a specific need to set the hostname here ? Usually, only port number is set.

Comment (emirsh) [reply cmd/hub-store/main.go]: Fixed this too

**Pull request merged**